extensions/2.0/Khronos/KHR_image_formation/README.md

```diff
@@ -0,0 +1,197 @@
+
+Supporting this extension is only appropriate for rendering a single glTF scene; since it defines a global output mapping it is impossible to respect conflicting mappings for two glTFs in the same scene. Likewise, if a shop wanted a consistent “look” across  their portfolio, they might be better served by picking a consistent mapping function and requiring their artists to design with this mapping in mind, rather than displaying different products with unique parameters. 
+
+This mapping function is split into two parts: the linear multiplier, exposure, and the nonlinear look-up tables (LUT). Exposure represents what the human pupil does to compensate for scenes with different average light levels, but here we will define it in terms from the photography world, as they have already created standard physical units to represent the camera’s version of the pupil. 
+
+The non-linear LUT is much more complicated and arbitrary, but is nevertheless completely necessary. This compression step is necessary because the [0, infinity) range of physical light must be converted to the [0, 1] range of any output format. When no LUT is specified, this generally means the value is clamped, which is simply a piecewise linear compression function with sharp corners that  create unsightly artifacts. This is rarely the default mode for existing renderers, as it yields such poor results, but their  defaults are also rarely consistent with each other, hence the value of this extension.
+
+There is no “right” way to compress outgoing scene light, but the goal is to mimic the nonlinear response of the human retina without a lot of expensive computation. Camera sensors and film are likewise nonlinear and designed also around the human eye’s response. This is the purpose of the hdr_lut. Effectively we are mapping from physical light units to unitless perception values, represented as sRGB or other wide-gamut, HDR formats. 
+
+One key aspect of a good transform is that it does not operate solely on luminance but also affects color, especially saturation. This is because as a colored light becomes very bright, its color ceases to be perceived. This is true of CCDs as well, and can be easily seen by taking an overexposed photo of a bright, colored light and noting how it tends toward white. This is the purpose of the tone_mapping_lut.
+
+Regarding color gamuts, since glTF defines its textures in sRGB, so long as the scene’s lighting is also represented in Rec 709 (sRGB’s color gamut), then the output light will naturally also be restricted to the Rec 709 gamut. If a wide gamut like Rec 2020 is used for output, then the values will naturally fall into the Rec 709 subset. 
```

Comment:
Would scene lighting with an AP1 gamut use the same HDR LUT as scene lighting in Rec 709? Or (more generally) do different LUTs impose restrictions on the scene linear color space?

Reply:
Unfortunately, different working color spaces would require different LUTs. In my own tests, I found little benefit to rendering in a wider gamut before the image formation step. It does help in specific cases (esp. when doing ray tracing with light bounces) but in glTF's current form, rendering in Rec.709 is sufficient

Reply:
> Would scene lighting with an AP1 gamut use the same HDR LUT as scene lighting in Rec 709?

To add to this, “wider gamut” is not “better” without defining a clear notion of what we are comparing.

### First Problem: Meaningfulness

RGB is simply tristimulus, and completely unrelated to how actual light transport works. We would be wise to accept that and move on. It’s really a “Good Enough” balance of bandwidth and computing, but ultimately anchored in human stimulus specifications.

Given that AP1 uses primaries that do not exist, we are faced with an additional “distance” to compress when we attempt to represent the stimulus in a medium. For example, because the AP1 primaries are meaningless with respect to the standard observer model that anchors our entire colourimetric work, even a BT.2020 idealized display with the pure spectral emitters would not represent anything like it. Because it is “beyond” the locus, it holds literally no meaning to the standard observer, and no speculation can be even made.

We can extend that question yo ask what the meaningless ratio means in terms of smaller gamuts?

Rendering something that has no meaning is part of the problem here.

### Second Problem: Choice of Rendering Tristimulus

A second problem, and one that is likely more dire, is that as we move outward to the spectrally pure values in a stimulus specification, we are _implicitly decreasing luminance_. Luminance can be considered _a portion_ of the general sense of “brightness” we perceive.

When we perform light transport on tristimulus of low luminance, the result is, unsurprisingly _dull as hell_. That is, counter to what people typically assume, the resulting mixtures _will appear less colourful_. The following examples are shamelessly borrowed from someone who [took it upon themselves to demonstrate the impact of chosen tristimulus space on resulting tristimulus renders and image formation, Chris Brejon](https://chrisbrejon.com/articles/ocio-display-transforms-and-misconceptions/)! Can you guess which of these were rendered a wider gamut and which were rendered in BT.709 based tristimulus?

<img width="1588" alt="Screen Shot 2022-03-08 at 8 49 06 AM" src="https://user-images.githubusercontent.com/59577/157286332-60dcc489-6d07-485f-bccc-ef22883a377c.png">

<img width="1588" alt="Screen Shot 2022-03-08 at 8 49 09 AM" src="https://user-images.githubusercontent.com/59577/157286174-74708565-4991-4dfe-851c-db127ddd80c0.png">

<img width="1588" alt="Screen Shot 2022-03-08 at 8 49 00 AM" src="https://user-images.githubusercontent.com/59577/157286378-d6da6020-8027-4cd6-b5f8-44c97f788ec8.png">

<img width="1588" alt="Screen Shot 2022-03-08 at 8 49 02 AM" src="https://user-images.githubusercontent.com/59577/157286356-ae2180ea-013f-425c-ac2f-ebecf7cf1613.png">

<img width="1588" alt="Screen Shot 2022-03-08 at 8 48 52 AM" src="https://user-images.githubusercontent.com/59577/157286432-3e2800b4-1f05-493e-bb7b-32b246dd8a49.png">

<img width="1588" alt="Screen Shot 2022-03-08 at 8 48 55 AM" src="https://user-images.githubusercontent.com/59577/157286408-3f9d30e3-8528-4f2b-8387-0d9a9f141ba3.png">

<img width="1588" alt="Screen Shot 2022-03-08 at 8 48 34 AM" src="https://user-images.githubusercontent.com/59577/157286610-43175e21-9f8a-48cb-88ea-400dce2c737b.png">

<img width="1588" alt="Screen Shot 2022-03-08 at 8 48 28 AM" src="https://user-images.githubusercontent.com/59577/157286853-bfdd3713-3d54-4c0a-8d62-b7745cffbd4a.png">

### Third Problem: "Gamut" Mapping

Even if we ignore the first two problems, we have a combination of the first problem present in our final output. If we are on a medium that cannot express the tristimulus value, such as going from BT.2020 to sRGB, how the heck do we formulate the result in the destination? A majority of approaches are just a simple clip, and that of course leads down the "hue" skew path of varying degrees, as well as making the result device dependent given that it will render differently on different output mediums.

Even if we aren't faced directly with the more abstract "meaningfulness" of nonsense values, we are again faced with the "meaning within the medium". For example, relative to sRGB, BT.2020 tristimulus mixtures may have meaning within the gamut, and no meaning for values that cannot be expressed. How do we give those values meaning? Do we focus on tonality so the plastic orange chair isn't a big huge flat wash of nasty? Do we try to decrease the perceptual facets of "chroma" or "hue" on our orange chair? What impact does that have when we go to the shop to buy the orange chair and it's... not the orange we thought we saw? These are pretty gnarly complexities that currently have no real answers without folks you to engineer them.

Sorry for the massive post, but I just wanted to get these things out into another public domain because there is frankly so much misunderstanding and false assumptions behind much of this rendering stuff that I think many folks could contribute toward solving if greater awareness were out there.

Reply:
Lots of news to me here, thanks @sobotka and @romainguy! 🙏

One takeaway I think I'm hearing (and please correct me if this is wrong) ... even when viewing the rendered image on a display that supports wider P3, AP1, or Rec 2020 gamuts, scene lighting calculated in a wider gamut working color space is not "better" (in any clear and simple sense) than scene lighting done in Rec 709?

Reply:
>  even when viewing the rendered image on a display that supports wider P3, AP1, or Rec 2020 gamuts, scene lighting calculated in a wider gamut working color space is not "better" (in any clear and simple sense) than scene lighting done in Rec 709?

Until we define what we are comparing on either side of the "better" the answer is yes. That is, in terms of image formation, having additional visual energy will lead to higher "colourfulness". Using pure primaries with exceptionally low luminance will potentially be detrimental here in terms of resulting image formed.

In the end, the analysis of "better" should be tempered against a clear declaration as to what is being compared, and why. It's an analysis of qualia, ultimately.